packages/lesswrong/components/ea-forum/giving-portal/ElectionCandidatesList.tsx

```diff
@@ -43,32 +60,81 @@ const sortOptions: Record<ElectionCandidatesSort, SettingsOption> = {
   },
 };
 
-const ElectionCandidatesList = ({className, classes}: {
+const sortOptions: Record<ElectionCandidatesSort | "random", SettingsOption> = {
+  ...selectSortOptions,
+  mostPreVoted: {
+    label: "Most pre-voted",
+  },
+};
+
+const ElectionCandidatesList = ({type="preVote", selectedCandidateIds, onSelect, setTotalCount, className, classes}: {
+  /**
+   * - "preVote": selecting a candidate (instantly) adds a pre-vote for it
+   * - "select": selecting a candidate runs the onSelect callback
+   */
+  type?: "preVote" | "select",
+  selectedCandidateIds?: string[],
+  onSelect?: (candidateIds: string[]) => void,
+  setTotalCount?: (count: number) => void,
   className?: string,
   classes: ClassesType,
 }) => {
-  const [sortBy, setSortBy] = useState<ElectionCandidatesSort>("mostPreVoted");
+  const isSelect = type === "select";
+  const [sortBy, setSortBy] = useState<ElectionCandidatesSort | "random">(
+    isSelect ? "random" : "mostPreVoted"
+  );
   const {results, loading} = useElectionCandidates(sortBy);
 
+  const allSelected = useMemo(
+    () => results?.every((candidate) => selectedCandidateIds?.includes(candidate._id)),
+    [results, selectedCandidateIds]
+  );
+
+  if (setTotalCount) {
+    setTotalCount(results?.length || 0);
+  }
+
   const onSelectSort = useCallback((value: string) => {
-    if (isElectionCandidateSort(value)) {
+    if (isElectionCandidateSort(value) || value === "random") {
       setSortBy(value);
     }
   }, []);
 
   const {Loading, ElectionCandidate, ForumDropdown} = Components;
   return (
     <div className={classNames(classes.root, className)}>
-      <ForumDropdown
-        value={sortBy}
-        options={sortOptions}
-        onSelect={onSelectSort}
-        className={classes.dropdown}
-      />
+      <div className={classes.controls}>
+        <ForumDropdown value={sortBy} options={isSelect ? selectSortOptions : sortOptions} onSelect={onSelectSort} className={classes.dropdown} />
```

Comment:
I'm leaning no because we don't use that consistently throughout the rest of the site already (see e.g. the core topics page), and I don't want to edit the default dropdown component just for this change

<img width="1127" alt="Screenshot 2023-11-28 at 22 37 55" src="https://github.com/ForumMagnum/ForumMagnum/assets/77623106/3c47670b-a2ed-41d6-a0ce-ebf54f4008ec">